packages/itwin/tree-widget/src/components/trees/stateless/models-tree/internal/ModelsTreeVisibilityHandler.ts

```diff
@@ -314,7 +314,16 @@ class ModelsTreeVisibilityHandlerImpl implements ModelsTreeVisibilityHandler {
 
       return from(this._idsCache.getModelCategories(modelId)).pipe(
         concatAll(),
-        map((categoryId) => this.getDefaultCategoryVisibilityStatus(categoryId, modelId)),
+        mergeMap((categoryId) => {
+          if (
+            viewport.perModelCategoryVisibility.getOverride(modelId, categoryId) === PerModelCategoryVisibility.Override.None &&
+            !viewport.view.viewsCategory(categoryId)
+          ) {
+            // If all category elements are always drawn it's actually visible.
+            return this.getCategoryDisplayStatus({ modelId, categoryId });
+          }
+          return of(this.getDefaultCategoryVisibilityStatus({ categoryId, modelId }));
```

Comment:
I don't understand why there's a special case for when the category is hidden through selector and there's no override. Can you explain?